### DISCUSSION TITLE
Whether I can use js/css framework to development the webview?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [microsoft/vscode-discussions#263] [Extension Development QnA] [answered]
When I study [webview](https://code.visualstudio.com/api/extension-guides/webview), 
I have several questions:

1. Whether I can use js/css framework to development the webview? such as `react`, `vue.js`, `jQuery`,  `bootstrap`,`TailwindCSS`.
2. Whether I can collect data of project using webview? 
such as, in a project, I can use webview to list all the dependencies of `node_modules` .
![image](https://user-images.githubusercontent.com/12727582/203757676-6abb85a8-b1ca-4532-b8e7-122bc9234c71.png)

**Comment by FredLL-Avaiga** [comment]
Of course, you can use js/css framework in a webview.
I would say that you'd need to have a separate project for the webview : different list of dependencies and a packager that builds a bundle that you 'll make available to the webview.
As you know, communication between vscode & the webview is done through messages

**Comment by r3m0t** [comment, marked as answer]
Tree views can be easier to program. There's an example here that is also showing node dependencies- https://github.com/microsoft/vscode-extension-samples/tree/main/tree-view-sample

**Reply by Tyriar** [reply]
This is also the recommended approach, you should only use a webview if you absolutely need to as they're much more difficult to work with and maintain the correct styling.
